pkg/services/ngalert/state/state.go

```diff
@@ -272,6 +294,8 @@ func resultError(state *State, rule *models.AlertRule, result eval.Result, logge
 	case models.ErrorErrState:
 		if state.State == eval.Error {
 			prevEndsAt := state.EndsAt
+			state.Error = result.Error
+			state.AddErrorAnnotations(result.Error, rule)
```

Comment:
This probably should be done in `case models.AlertingErrState` too, but I think we can do it separately.